### PR TITLE
feat: UNT-T23127: Ripple Button With Animated API

### PR DESCRIPTION
**Pull request by nilesh-simform**
- Updated Peer Dependency
- Added Ripple Effect Button With Animated API

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,6 +96,44 @@ const styles = StyleSheet.create({
...
 
+
+Example of Ripple Effect Button
+
+```jsx
+import React from 'react';
+import {StyleSheet, Text} from 'react-native';
+import {RippleButton} from 'react-native-spinner-button';
+
+const MyRippleButton: React.FC = () => {
```
Comment (mukesh-simform) [inline README.md:107]: @nilesh-simform We should rename this MyRippleButton to CustomButton

File: README.md [hunk 2]
```diff
@@ -96,6 +96,44 @@ const styles = StyleSheet.create({
...
+  const buttonPress: () => void = () => {
+    console.log('Button Clicked');
+  };
+
+  return (
+    <RippleButton
+      onPress={buttonPress}
+      animatedDuration={800}
+      rippleColor="rgba(0,0,0,0.3)">
```
Comment (mukesh-simform) [inline README.md:116]: @nilesh-simform  we should format this correctly

File: src/components/RippleButton/RippleButton.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import styles from './RippleStyles';
+
+const RippleButton: React.FC<RippleProps> = ({
+  style,
+  onPress,
+  animatedDuration = 500,
+  rippleColor = 'rgba(0,0,0,0.2)',
+  children,
+}) => {
+  const translation: Animated.Value = useRef(new Animated.Value(0)).current;
```
Comment (mukesh-simform) [inline src/components/RippleButton/RippleButton.tsx:19]: @nilesh-simform We should create a custom hook inside RippleButton to move this animation logic there.
e.g RipplerButton/useRippleButton.ts

File: src/components/RippleButton/RippleStyles.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { StyleSheet } from 'react-native';
+
+export default StyleSheet.create({
```
Comment (mukesh-simform) [inline src/components/RippleButton/RippleStyles.ts:3]: @nilesh-simform It should be RippleButtonStyles.ts

**Review by mukesh-simform — DISMISSED** [review]

File: src/components/Button/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+const Button: React.FC<ButtonProps> = ({
+  animationType,
+  style,
+  onPress,
+  animatedDuration = 500,
+  rippleColor = 'rgba(255, 255, 255, .25)',
+  children,
+}) => {
+  switch (animationType) {
```
Comment (Krunal-K-SimformSolutions) [inline src/components/Button/Button.tsx:13]: @nilesh-simform Why add a switch because in both cases use the same code?
CC. @mukesh-simform

**Review by mukesh-simform — APPROVED** [review]

**Pull request merged**